### DISCUSSION TITLE
Troubleshooting Kubernetes Gateway API configuration with Contour on MicroK8s

### DISCUSSION THREAD

**Discussion by ppb31** [projectcontour/contour#6296] [Q&A]
Hello everyone, 

I am currently experiencing difficulties configuring the Kubernetes Gateway API with Contour on my MicroK8s cluster (version 1.26). I installed MicroK8s using the following command:

sudo snap install microk8s --classic --channel=1.26/stable

Then, I deployed Contour following the official documentation and using the provided YAML manifests. However, after creating the GatewayClass and Gateway resources, I notice that the Gateway remains stuck in a "Waiting for controller" state with an "Unknown" status. Here are the results of the commands used to check the status of the resources:

    GatewayClass status:

    kubectl get gatewayclass 
    NAME      CONTROLLER                  ACCEPTED   AGE
    contour   projectcontour.io/contour   Unknown    11s

Gateway status:

kubectl get gateway -n projectcontour
NAME              CLASS     ADDRESS   PROGRAMMED   AGE 
contour-gateway   contour             Unknown      18m

Detailed Gateway description:

kubectl describe gateway -n projectcontour contour-gateway
...
Status:
  Conditions:
    Last Transition Time:  1970-01-01T00:00:00Z
    Message:               Waiting for controller
    Reason:                Pending  
    Status:                Unknown
    Type:                  Accepted
    Last Transition Time:  1970-01-01T00:00:00Z
    Message:               Waiting for controller
    Reason:                Pending
    Status:                Unknown 
    Type:                  Programmed
Events:                    <none>

The Contour and Envoy pods are running without any apparent errors, and the Contour logs do not show any specific errors related to the Gateway. It's important to note that I encountered similar issues when using Traefik as the Gateway API controller, which suggests the problem is likely not specific to Contour or Traefik. I would appreciate any suggestions or leads to unblock this situation and get the Kubernetes Gateway API working with Contour on my MicroK8s cluster. 

Thank you in advance for your help!

**Comment by sunjayBhatia** [comment]
Hey @ppb31 what method for installing Contour did you use, static provisioning or dynamic provisioning with the Gateway Provisioner? (see: https://projectcontour.io/docs/main/config/gateway-api/#enabling-gateway-api-in-contour) If "static," what does your contour configuration ConfigMap look like? You'll need to make sure Contour is configured to reconcile either a specific Gateway or (now-deprecated on main) GatewayClasses with a particular controller name
